### PR TITLE
[DRAFT] partial impl of sox environ check

### PR DESCRIPTION
**Pull request by robbintt**

### CODE REVIEW

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java [hunk 1]
```diff
@@ -235,6 +236,13 @@ public Response create(
             }
         }
 
+        // only allow deploy of sox if not private and from sox_compliant source
```
Comment (ankilosaurus) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java:239]: can be moved to helper method.

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java [hunk 2]
```diff
@@ -227,6 +227,7 @@ public Response create(
                     buildBean.getArtifact_url()));
         }
         // if the stage is not allowed (allow_private_build)
+        // TODO: remove this code once is_sox db column fully populated (before merging this)
```
Comment (ankilosaurus) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java:230]: Should have followup jira for tracking this?

File: deploy-service/common/src/main/java/com/pinterest/deployservice/allowlists/BuildAllowlistImpl.java [hunk 3]
```diff
@@ -35,4 +37,14 @@ public Boolean trusted(String buildName) {
         }
         return false;
     }
+
+    // sox_compliant checks if build matches trusted URL allow list
+    public Boolean sox_compliant(String buildName) {
+        for (String pattern : soxBuildURLs) {
+            if (buildName.matches(pattern)) {
```
Comment (ankilosaurus) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/allowlists/BuildAllowlistImpl.java:44]: can buildNamebe null? same as other places though.

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java (hunk 2 above)
Comment (robbintt) [reply deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java:230]: Good idea, done.

File: deploy-service/common/src/main/java/com/pinterest/deployservice/allowlists/BuildAllowlistImpl.java (hunk 3 above)
Comment (robbintt) [reply deploy-service/common/src/main/java/com/pinterest/deployservice/allowlists/BuildAllowlistImpl.java:44]: It's not nullable in the db schema, thanks for checking

**Comment by robbintt** [comment]
merged 2 PRs that shared getters & setters

**Pull request closed without merging**